### DISCUSSION TITLE
Wright Fisher forward time simulation with discrete generation feature

### DISCUSSION THREAD

**Discussion by NiharaDeSilva** [tskit-dev/msprime#2254] [General]
I am trying to use msprime to simulate populations. I would like to know whether msprime supports forward time simulation of  Wright-Fisher model along with the discrete generation feature where i could simulate the populations for a given duration range(eg: 2,8) . According to the documentation of msprime, it supports discrete backwards-time Wright-Fisher model. However I have written the following code to simulate populations with discrete forward time Wright-Fisher model 
Really appreciate if this could be clarified.

tree_sequence = msprime.sim_ancestry(samples=sample_size, model=”dtwf”, ploidy=2, recombination_rate=1e-8, population_size=effective_population, sequence_length=loci, random_seed=42)
tree_sequence = msprime.sim_mutations(tree_sequence, start_time=2, end_time=8, rate=rate, random_seed=42)

**Comment by GertjanBisschop** [comment]
Hi @NiharaDeSilva,
Thanks for reaching out. The code snippet you posted here will simulate the genealogical history for your sample under Wright-Fisher dynamics until all samples have reached a local common ancestor (the `sim_ancestry` part). The Wright-Fisher model is simulated backwards in time by msprime, however, there should not be any statistical differences between doing it forwards or backwards. As you say, this model does all steps on a per generation basis and is therefore computationally more expensive than the coalescent. This is why different models are often combined, as further back in time the coalescent is a good enough approximation anyway. See [this section](https://tskit.dev/msprime/docs/stable/ancestry.html#combining-backward-time-simulations) on how to combine different models. There are some code examples there.

In the next line of code you sprinkled mutations along the branches of all local trees. Applying a start and end time here will therefore only affect the mutation process. (See the documentation linked above on how to set the duration on the `sim_ancestry` part instead.)

I hope this clarifies at least some of the questions you had.

**Reply by NiharaDeSilva** [reply]
Hi @GertjanBisschop, Thank you for the clarifications provided. I have corrected the code to simulate populations using only the Wright-Fisher discrete model for generation range (2,8) and have mutations. (disregarding the computational cost involved).  Hope this is reasonable.
tree_sequence = msprime.sim_ancestry(samples=sample_size, start_time=2, model=msprime.DiscreteTimeWrightFisher(duration=6),  ploidy=2, recombination_rate=1e-8, population_size=effective_population, sequence_length=loci, random_seed=42)
tree_sequence = msprime.sim_mutations(tree_sequence, rate=mutation_rate, random_seed=42)

**Reply by GertjanBisschop** [reply]
Glad my comments were helpful. This bit of code will run the Wright-Fisher model indeed for 6 generations and will then stop (regardless of whether all samples have a single common ancestor in each of the marginal trees). No random events will be simulated between time 0 and 2 (`start_time=2`).
